.github/workflows/build.yml

```diff
@@ -128,3 +128,171 @@ jobs:
 
       - name: Run make kotlin
         run: make -C lib/kotlin
+
+  lib-rust:
+    needs: compiler
+    runs-on: ubuntu-20.04
+    env:
+      TOOLCHAIN_VERSION: 1.40.0
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -yq
+          sudo apt-get install -y --no-install-recommends $BUILD_DEPS
+          sudo apt-get install -y --no-install-recommends curl
+
+      - name: Setup cargo
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          rustup update
+          rustup install $TOOLCHAIN_VERSION
+          rustup default $TOOLCHAIN_VERSION
+          rustup --version
+          cargo --version
+          rustc --version
+
+      - name: Run bootstrap
+        run: ./bootstrap.sh
+
+      - name: Run configure
+        run: |
+          ./configure \
+            --disable-debug \
+            --disable-tests \
+            --disable-dependency-tracking \
+            --without-cpp \
+            --without-c_glib \
+            --without-java \
+            --without-kotlin \
+            --without-python \
+            --without-py3 \
+            --without-ruby \
+            --without-haxe \
+            --without-netstd \
+            --without-perl \
+            --without-php \
+            --without-php_extension \
+            --without-dart \
+            --without-erlang \
+            --without-go \
+            --without-d \
+            --without-nodejs \
+            --without-nodets \
+            --without-lua \
+            --with-rs \
+            --without-swift
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: thrift-compiler
+          path: compiler/cpp
+
+      - name: Run thrift-compiler
+        run: |
+          chmod a+x compiler/cpp/thrift
+          compiler/cpp/thrift -version
+
+      - name: Run make rust
+        run: make -C lib/rs
+
+  lib-python:
+    needs: compiler
+    runs-on: ubuntu-20.04
```

Comment:
should we also make this part of the matrix? my understanding is that this is not the only version of ubuntu we want to support.

Reply:
i'm adding matrix to the compiler part but for python i think it's okay to have just one - given how libraries and compilers are in many cases compiled, installed, and used separately. the matrix should be focusing on multiple python versions for this part.